src/plugins/info/plugin.py

```diff
@@ -112,18 +170,57 @@ def on_config(self, config):
         example, _ = os.path.splitext(example)
         example = "-".join([present, slugify(example, "-")])
 
+        # Load exclusion patterns
+        self.exclusion_patterns = _load_exclusion_patterns()
+
+        # Exclude the site_dir at project root
+        if config.site_dir.startswith(os.getcwd()):
+            self.exclusion_patterns.append(_resolve_pattern(config.site_dir))
+
+        # Exclude the site-packages directory
+        for path in site.getsitepackages():
+            if path.startswith(os.getcwd()):
+                self.exclusion_patterns.append(_resolve_pattern(path))
+
+        # Exclude site_dir for projects
+        if projects_plugin:
+            for path in glob.iglob(
+                pathname = projects_plugin.config.projects_config_files,
+                root_dir = abs_projects_dir,
+                recursive = True
+            ):
+                current_config_file = os.path.join(abs_projects_dir, path)
+                project_config = _get_project_config(current_config_file)
+                pattern = _resolve_pattern(project_config.site_dir)
+                self.exclusion_patterns.append(pattern)
+
         # Create self-contained example from project
         files: list[str] = []
         with ZipFile(archive, "a", ZIP_DEFLATED, False) as f:
-            for path in ["mkdocs.yml", "requirements.txt"]:
-                if os.path.isfile(path):
+            for abs_root, dirnames, filenames in os.walk(os.getcwd()):
+                # Prune the folders in-place to prevent
+                # scanning excluded folders
+                for name in list(dirnames):
+                    path = os.path.join(abs_root, name)
+                    if self._is_excluded(_resolve_pattern(path)):
+                        dirnames.remove(name)
+                        continue
+                    # Multi-language setup from #2346 separates the
+                    # language config, so each mkdocs.yml file is
+                    # unaware of other site_dir directories. Therefore,
+                    # we add this with the assumption a site_dir contains
+                    # the sitemap file.
+                    sitemap_gz = os.path.join(path, "sitemap.xml.gz")
+                    if os.path.exists(sitemap_gz):
+                        log.debug(f"Excluded site_dir: {path}")
+                        dirnames.remove(name)
+                for name in filenames:
```

Comment:
Please add some more inline comments that make it easier to understand what is happening in which order. This might seem pedantic, but will make maintenance down the line much simpler.